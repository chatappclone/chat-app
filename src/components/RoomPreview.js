import React from 'react';
import '../styles/RoomPreview.scss';

class RoomPreview extends React.Component {

  constructor () {
    super()

  this.handleClick = this.handleClick.bind(this);
  }

handleClick () {
  this.props.receiveHandleCurrentRoom(this.props.room)
}

  render() {

    return (
      <div onClick={this.handleClick}>
        <h4>{this.props.room.name}</h4>
        <p>{this.props.room.updated_at}</p>
      </div>
    )
  }
}
export default RoomPreview;
