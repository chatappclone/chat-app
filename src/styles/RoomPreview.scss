@import 'defaults';

.room-preview {
  display: flex;
  justify-content: space-around;
  align-items: center;
  max-height: 80px;
}

.room-preview__img {
  border-radius: 50%;
  max-width: 40px;
  max-height: auto;
}

.room-preview__preview-info {
  border-bottom: 1px solid grey;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.room-preview__room-details {

}
.room-preview__name {
  display: inline-block;
}
