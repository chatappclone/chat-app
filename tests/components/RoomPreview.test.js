import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RoomPreview from '../../src/components/RoomPreview';

const room = {};
const roomMap = [];
const receiveHandleCurrentRoom = jest.fn();

describe('Room Preview', () => {
  test('matches the snapshot', () => {
    const tree = renderer
      .create(
        <RoomPreview
          room={room}
          roomMap={roomMap}
          receiveHandleCurrentRoom={receiveHandleCurrentRoom}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('click calls function', () => {
    const wrapper = shallow(
      <RoomPreview
        room={room}
        roomMap={roomMap}
        receiveHandleCurrentRoom={receiveHandleCurrentRoom}
      />,
    );
    wrapper.find('.room-preview').simulate('click');
    expect(receiveHandleCurrentRoom).toHaveBeenCalled();
  });
});
