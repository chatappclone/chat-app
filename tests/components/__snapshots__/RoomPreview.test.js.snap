// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room Preview matches the snapshot 1`] = `
<div
  className="room-preview"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="room-preview__avatar"
  >
    0
  </div>
  <div
    className="room-preview__preview-info"
  >
    <div
      className="room-preview__room-name"
    >
      0
      <p>
        NaN:NaN
      </p>
    </div>
    <div
      className="room-preview__preview-text"
    >
      <p>
        <span
          className="tick"
        >
          <svg
            height="15"
            id="msg-dblcheck-ack"
            width="16"
            x="2063"
            xmlns="http://www.w3.org/2000/svg"
            y="2076"
          >
            <path
              d="M15.01 3.316l-.478-.372a.365.365 0 0 0-.51.063L8.666 9.88a.32.32 0 0 1-.484.032l-.358-.325a.32.32 0 0 0-.484.032l-.378.48a.418.418 0 0 0 .036.54l1.32 1.267a.32.32 0 0 0 .484-.034l6.272-8.048a.366.366 0 0 0-.064-.512zm-4.1 0l-.478-.372a.365.365 0 0 0-.51.063L4.566 9.88a.32.32 0 0 1-.484.032L1.892 7.77a.366.366 0 0 0-.516.005l-.423.433a.364.364 0 0 0 .006.514l3.255 3.185a.32.32 0 0 0 .484-.033l6.272-8.048a.365.365 0 0 0-.063-.51z"
              fill="#4fc3f7"
            />
          </svg>
        </span>
      </p>
    </div>
  </div>
  <div
    className="room-preview__chevron"
  >
    <i
      className="zmdi zmdi-chevron-right"
    />
  </div>
</div>
`;
